pkg/controller/stack/stack_controller.go

```diff
@@ -1556,20 +1626,46 @@ func (sess *reconcileStackSession) InstallProjectDependencies(ctx context.Contex
 }
 
 func (sess *reconcileStackSession) UpdateConfig(ctx context.Context) error {
-	m := make(auto.ConfigMap)
+	// Initialize a single config value slice to all values;
+	// plain config will be handled with values from ConfigRefs
+	configValues := make([]ConfigKeyValue, 0, len(sess.stack.Config))
+
 	for k, v := range sess.stack.Config {
-		m[k] = auto.ConfigValue{
-			Value:  v,
-			Secret: false,
-		}
+		configValues = append(configValues, ConfigKeyValue{
+			Key: k,
+			Value: auto.ConfigValue{
+				Value:  v,
+				Secret: false,
+			},
+		})
+	}
+
+	// appending ConfigRefs values
+	configValuesFromRefs, err := sess.resolveConfigRefs(ctx)
+	if err != nil {
+		return fmt.Errorf("Fail reading ConfigRef values: %w", err)
 	}
+	configValues = append(configValues, configValuesFromRefs...)
+
+	// config values should be ordered by key to avoid problems like try to set 'c[1]=value1' before 'c[0]=value2'
+	sort.Slice(configValues, func(i, j int) bool {
+		return configValues[i].Key < configValues[j].Key
+	})
```

Comment:
Is the second example a valid syntax? as a json?

Reply:
I just doubled checked -- these are two ways of describing the same thing:

```
  stack:deepobject:
    nums:
      - 1
  stack:flatobject: '{"nums": [1]}
```

It doesn't look like `stack:object.nums[0]: 1` is valid.

I think it would be totally fine to say this is an edge case we don't handle, given that it's unlikely to come up.